x-pack/plugins/maps/public/classes/joins/inner_join.ts

```diff
@@ -52,12 +66,18 @@ export class InnerJoin {
     return `${this.getSourceDataRequestId()}_${FORMATTERS_DATA_REQUEST_ID_SUFFIX}`;
   }
 
-  getLeftField() {
+  getLeftField(): IField {
+    if (!this._leftField) {
+      throw new Error('Cannot get leftField from InnerJoin with incomplete config');
+    }
     return this._leftField;
   }
 
-  joinPropertiesToFeature(feature, propertiesMap) {
-    const rightMetricFields = this._rightSource.getMetricFields();
+  joinPropertiesToFeature(feature: Feature, propertiesMap: PropertiesMap): boolean {
+    if (!feature.properties || !this._leftField) {
```

Comment:
Check for rightFields here and return false if not provided instead of using not null operator on line 80